### PR TITLE
Jetpack Manage: Show payment notice in site payment methods page

### PR DESCRIPTION
**Pull request by yashwin**
Resolves https://github.com/Automattic/jetpack-genesis/issues/192

## Proposed Changes

This PR implements a notice that shows up in the payment methods page in the site details page in Jetpack Cloud.

## Testing Instructions

**Prerequisites**

Since these changes are made specifically for agencies, you must set yourself(partner) as an agency - 2c49b-pb.

**Instructions**

1. Visit the Jetpack Cloud live link > You will be redirected to /dashboard
2. Click `All Sites` to switch sites and select a site > Click the `Purchases` left nav item
3. Now click the Payment Methods tab and verify that a notice is displayed as shown below and verify that clicking the `managed here` link takes you to `/partner-portal/payment-methods`

Screenshot 2024-01-18 at 3 49 16 PM

4. Now, click the Calypso live link > Visit /me/purchases/payment-methods and verify that the notice is no more shown:

Screenshot 2024-01-18 at 11 44 56 AM

5. Verify that you can add a credit card

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/me/purchases/payment-methods/payment-method-list.tsx
Comment (jkguidaven) [inline client/me/purchases/payment-methods/payment-method-list.tsx]: Should we add another `isAgency` check to ensure it is only visible to agency users?

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10834225

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @yashwin for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/10834225).

File: client/me/purchases/payment-methods/payment-method-list.tsx
Comment (yashwin) [reply client/me/purchases/payment-methods/payment-method-list.tsx]: Very good point. I have made the changes: https://github.com/Automattic/wp-calypso/pull/86572/commits/623d4bb01d4b20c9d60658ac5447883e9491f9ec

**Comment by yashwin** [comment]
Thanks for the review, @jkguidaven!

I have addressed your comments. Could you please take a look at this again? Thanks!

I have also updated the testing instructions.

**Review by jkguidaven — APPROVED** [review]
Thank you for addressing my comment. I retested the functionality using both agency and non-agency accounts and can confirm that it works as expected.  👍 

Before merging, I suggest getting approval from either Simon or Design to ensure the copy makes sense.

**Comment by yashwin** [comment]
> Thank you for addressing my comment. I retested the functionality using both agency and non-agency accounts and can confirm that it works as expected. 👍

Thanks for the review, @jkguidaven!

> Before merging, I suggest getting approval from either Simon or Design to ensure the copy makes sense.

The text was given by Simon and was approved by Jeff. You can check the details here: pedEI5-1dY-p2#comment-862. However, I have already asked for a review here. Will ping them for the approval. Thanks!

**Review by simonktnga8c — APPROVED** [review]
Works great 👍

**Pull request merged**
